Library/brew.sh

```diff
@@ -72,13 +76,47 @@ then
   fi
 fi
 
-export HOMEBREW_VERSION
+HOMEBREW_CURL="/usr/bin/curl"
+HOMEBREW_PROCESSOR="$(uname -p)"
+if [[ -n "$HOMEBREW_OSX" ]]
+then
+  HOMEBREW_PRODUCT="Homebrew"
+  HOMEBREW_SYSTEM="Macintosh"
+  # This is i386 even on x86_64 machines
+  [[ "$HOMEBREW_PROCESSOR" = "i386" ]] && HOMEBREW_PROCESSOR="Intel"
+  HOMEBREW_OSX_VERSION="$(/usr/bin/sw_vers -productVersion)"
+  HOMEBREW_OS_VERSION="Mac OS X $HOMEBREW_OSX_VERSION"
+
+  HOMEBREW_OSX_VERSION_NUMERIC="$(printf "%02d%02d%02d" $(echo "$HOMEBREW_OSX_VERSION" | tr '.' ' '))"
```

Comment:
You could use `"${HOMEBREW_OSX_VERSION//./ }"` to avoid echo and tr. Also may be just 
```bash
if [[ "$(printf "%02d%02d%02d" "${HOMEBREW_OSX_VERSION//./ }")" -lt "100900" && \
   -x "$HOMEBREW_PREFIX/opt/curl/bin/curl"  ]]
```
This is to avoid unnecessary variable HOMEBREW_OSX_VERSION_NUMERIC